src/transactions/simulation/TxSimUtils.cpp

```diff
@@ -109,6 +109,50 @@ generateScaledOfferID(OperationResult const& result, uint32_t partition)
     return generateScaledOfferID(offerID, partition);
 }
 
+Hash
+generateScaledClaimableBalanceID(OperationResult const& result,
+                                 uint32_t partition)
+{
+    if (result.code() != opINNER)
+    {
+        // This will produce txINTERNAL_ERROR on apply
+        throw std::runtime_error(
+            "generateScaledOfferID: invalid OperationResult");
+    }
+
+    if (result.tr().type() != CREATE_CLAIMABLE_BALANCE)
+    {
+        throw std::runtime_error(
+            "Invalid OperationResult: must be CREATE_CLAIMABLE_BALANCE");
+    }
+
+    auto const& balanceID =
+        result.tr().createClaimableBalanceResult().balanceID().v0();
+    return generateScaledClaimableBalanceID(balanceID, partition);
+}
+
+Hash
+generateScaledClaimableBalanceID(Hash const& balanceID, uint32_t partition)
+{
+    if (partition == 0)
```

Comment:
Nitpick: I'm not sure how much this special case really matters. There is no problem with leaving it here, but it's probably marginal.

Reply:
Good point. Removed the case to make this simpler.